extras/videoDrivers/SDL2/VideoSDL2.cpp

```diff
@@ -213,29 +213,27 @@ void VideoSDL2::PrintError(const std::string& msg) const
 
 void VideoSDL2::HandlePaste()
 {
-#ifdef _WIN32
-    if(!IsClipboardFormatAvailable(CF_UNICODETEXT))
+    if(!SDL_HasClipboardText())
         return;
 
-    OpenClipboard(nullptr);
-
-    HANDLE hData = GetClipboardData(CF_UNICODETEXT);
-    const wchar_t* pData = (const wchar_t*)GlobalLock(hData);
+    auto* text = SDL_GetClipboardText();
+    if(!text)
+        return;
+    if(text[0] == '\0')
+        PrintError(SDL_GetError());
 
     KeyEvent ke = {KeyType::Invalid, 0, false, false, false};
-    while(pData && *pData)
+    auto* p = text;
+    while(*p)
     {
-        ke.c = *(pData++);
+        ke.c = static_cast<unsigned char>(*(p++));
```

Comment:
The result of `SDL_GetClipboardText` seems to be UTF-8 encoded. So you need to decode each codepoint here.

You can use `s25util::utf8to32`, as in the `SDL_TEXTINPUT` handling especially as I don't think we should have handled space differently when pasting.

Reply:
Oh, I didn't know that was handled at this level. Now the `int` type makes sense.

Reply:
I just double-checked, that pasting a space still works. :+1:

Reply:
> Oh, I didn't know that was handled at this level. Now the int type makes sense.

by now we should use `char32_t` for that member (`c`), but not sure if anything would break by that.